lib/cart/models/cart_message.dart

```diff
@@ -0,0 +1,21 @@
+enum CartMessageType {
+  success,
+  error,
+  warning,
+  info,
+}
+
+abstract class CartMessage {
+  final CartMessageType type;
+  final String message;
```

Comment:
We don't really use those yet, but it seems we are mixing UI and state management here 🙂 

Message are "events", so it should be a simple description of what happened. The class name is sufficient. What to display and how to render (success, warning, etc.) is a different matter.

(btw I'm ok with the "message" semantic to avoid confusion with the bloc event)

Reply:
Oh maybe you were thinking about the logging system? not the user feedback?

In that case, nevermind 👍

Reply:
That's correct. In these cases, I was thinking about a psuedo-event bus architecture where messages can be sent out for multiple purposes (UI, analytics, logging, etc.) so other components can update themselves or handle accordingly.